dev/RadioButtons/ColumnMajorUniformToLargestGridLayout.cpp

```diff
@@ -0,0 +1,183 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+#include "common.h"
+#include "ResourceAccessor.h"
+#include "Utils.h"
+#include "ColumnMajorUniformToLargestGridLayout.h"
+
+CppWinRTActivatableClassWithBasicFactory(ColumnMajorUniformToLargestGridLayout);
+
+// IItemsControlOverrides
+
+ColumnMajorUniformToLargestGridLayout::ColumnMajorUniformToLargestGridLayout()
+{
+}
+
+winrt::Size ColumnMajorUniformToLargestGridLayout::MeasureOverride(
+    winrt::NonVirtualizingLayoutContext const& context,
+    winrt::Size const& availableSize)
+{
+    if (auto const children = context.Children())
+    {
+        auto const maxColumns = std::max(1, MaximumColumns());
+        MUX_ASSERT(maxColumns > 0);
+        auto const maxItemsPerColumn = static_cast<int>(std::ceil(static_cast<double>(children.Size()) / static_cast<double>(maxColumns)));
+
+        for (auto const child : children)
+        {
+            child.Measure(availableSize);
+        }
+        auto const largestChildSize = LargestChildSize(context);
+
+        auto const actualColumnCount = static_cast<float>(std::min(
+            static_cast<double>(maxColumns),
+            static_cast<double>(children.Size())));
+        return winrt::Size(
+            (largestChildSize.Width * actualColumnCount) + 
+            (ColumnSpacing() * (actualColumnCount - 1)),
```

Comment:
Row and ColumnSpacing not exposed publicly ?

Reply:
They are exposed in the layouts IDL, they aren't exposed via radio buttons.

---
In reply to: [347413392](https://github.com/microsoft/microsoft-ui-xaml/pull/1623#discussion_r347413392) [](ancestors = 347413392)